admin/cluster_containers_setup.py

```diff
@@ -0,0 +1,324 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+import yaml
+import os
+import sys
+
+from copy import deepcopy
+from json import dumps
+from itertools import repeat
+from treq import json_content
+
+from twisted.internet.defer import inlineCallbacks
+from twisted.python.filepath import FilePath
+from twisted.python import usage
+from twisted.web.http import OK
+
+from flocker.common import loop_until
+from flocker.control.httpapi import REST_API_PORT
+from flocker.apiclient import FlockerClient
+from flocker.ca import treq_with_authentication
+
+
+class ContainerOptions(usage.Options):
+    """
+    Parses the options pased as an argument to the create container script.
+    """
+    # XXX mandatory options
+    # XXX validations
+    description = "Set up containers in a Flocker cluster."
+
+    optParameters = [
+        ['apps-per-node', None, 0, 'Number of application containers per node',
+         int],
+        ['app-template', None, None,
+         'Configuration to use for each application container'],
+        ['control-node', None, None,
+         'Public IP address of the control node'],
+        ['cert-directory', None, None,
+         'Location of the user and control certificates and user key'],
+        ['wait', None, None,
+         "The timeout in seconds for waiting until the operation is complete. "
+         "No waiting is done by default."],
+        ['cluster', None, None,
+         'Configuration of the cluster'],
+    ]
+
+    synopsis = ('Usage: setup-cluster-containers --app-per-node <containers'
+                'per node> --app-template <name of the file> '
+                '--control-node <IPAddress>'
+                '--cert-directory <path where all the certificates are>'
+                '[--cluster <file> (UNUSED)')
```

Comment:
Don't have an unused option in new code. If this is for future work, add it when it does something.